modules/authentication/Interceptors.js

```diff
@@ -15,10 +15,10 @@ class Interceptors {
     return this.store
       .getHydratedState()
       .then((state) => {
-        const tokenInfoFields = Object.keys(state.authentication.tokenInfo);
+        const tokenInfoFields = state.getIn(['authentication', 'tokenInfo']).keySeq().toArray();
```

Comment:
Any reason to do `keySeq` instead of just `keys`?

Reply:
@dkniffin Good question. I believe when setting this up on ROAMD I had initially copied the use of `keySeq` from somewhere. However, I've dug a little further into this and I'll outline what I've come up with.

From what I've gathered, using `keys` returns an `Iterator` while `keySeq` returns an `Iterable` and both have different APIs. They attempted to match the `Iterator` protocals (which can be found [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols)) for `keys` which gives you things like using `next` but for Immutable doesn't give you the `Sequence` API.

See some more info in the links below:
* https://github.com/facebook/immutable-js/issues/529
* https://stackoverflow.com/questions/35623347/immutable-js-get-keys-from-map-hash

But basically if you use `keys` you can't then also do an `toArray` at the end.

![state](https://user-images.githubusercontent.com/10897808/36383537-684c4098-155a-11e8-83e2-3df9fe7b87cc.png)

Reply:
for what its worth, `keySeq` also maintains the orders of the keys, whereas `keys` does not. Personally I've found `keySeq`, 'valueSeq`, and 'entrySeq` to behave much more practicably than their non-`seq` counterparts.